text/3374-cargo-target-features.md

```diff
@@ -0,0 +1,212 @@
+* A new CLI argument could be added to change the behavior of `required-features` to behave the same as `enable-features`, avoiding the need to add `enable-features`.
+  This is viable, but the goal of this RFC is to make it as easy as possible to work with the cargo targets without requiring additional CLI arguments.
+
+## Alternate workflows
+
+* Instead of using `enable-features`, developers can be diligent in passing the appropriate `--features` options on the command-line when building their projects, possibly using `required-features` to ensure they only get built in the correct scenarios.
+  The intent of this RFC is to make that process easier and more seamless.
+  The current UX for using `required-features` can be confusing when the required features are not specified.
+  For example, `cargo install` fails to inform why it fails (see [#11617](https://github.com/rust-lang/cargo/issues/11617)).
+* Users can set up [aliases](https://doc.rust-lang.org/cargo/reference/config.html#alias) which pass in the feature flags they want to enable.
+  This can help with a development workflow, but requires more documentation and education, and doesn't help with some commands like a remote `cargo install`.
+* Developers can organize their project in a [Cargo Workspace](https://doc.rust-lang.org/cargo/reference/workspaces.html) instead of using multiple targets within a single package.
+  This allows customizing dependencies and other settings within each package.
+  Workspaces can add some more overhead for managing multiple packages, but offer a lot more flexibility.
+  Instead of implementing this RFC, we could put more work into making workspaces easier to use, which could benefit a larger audience.
+  However, it is not clear if it is feasible to make workspaces work as effortlessly compared to targets.
+  Also, there is likely more work involved to get workspaces on the same level.
+
+## Alternate designs
+
+* [RFC 2887](https://github.com/rust-lang/rfcs/pull/2887) proposes being able to add dependency tables directly in a target definition.
+  It is intended that this RFC using `enable-features` will hopefully be easier to implement, make it easier to reuse a dependency declaration across multiple targets, and allow working with features that are not related to dependencies.
+* [RFC 3020](https://github.com/rust-lang/rfcs/pull/3020) proposes an enhancement similar to this RFC.
+
+## Other alternative considerations
+
+* [cargo#1982](https://github.com/rust-lang/cargo/issues/1982) is the primary issue requesting the ability to set per-target dependencies, and contains some discussion of the desired use cases.
+* Other names may be considered for the field `enable-features`, such as `forced-features`, `force-enable-features`, etc.
+* A flag could be added to the target definition to indicate that default features should be disabled when the target is being built.
```

Comment:
This is not exactly an alternative but an extension on top of the RFC.

I think this should be actually be inside this RFC itself, but if not I will create a quick follow up RFC since IMHO, there are good use cases for this.

The flag name could be:

```toml
disable-default-features = true
```